tiptapy/extras.py

```diff
@@ -8,15 +8,19 @@ class FeatureImage(BaseNode):
     def inner_render(self, node) -> str:
         special_attrs_map = {'caption': 'figcaption'}
         attrs = node.get("attrs", {})
-        attrs_s = " ".join(f'{k}="{v}"'
-                           for k, v in attrs.items()
-                           if k not in special_attrs_map
-                           )
-        html = f"<picture><img {attrs_s}><picture>"
-        for attr in special_attrs_map:
-            if attr in attrs:
-                tag = special_attrs_map[attr]
-                html += f"<{tag}> {attrs[attr]} </{tag}>"
+        attrs_s = " ".join(
+            f'{k}="{v}"'
+            for k, v in attrs.items()
+            if k not in special_attrs_map and
+            len(attrs[k]) != 0
+        )
+        html = f"<picture><img {attrs_s}></picture>"
+        # Checking if caption is not empty
+        if len(attrs['caption']) != 0:
```

Comment:
@Pradhvan `for` loop won't get executed for other special attributes also. is it a correct behavior? cc: @shon

Reply:
The only special character is `<figcaption></figcaption>` else everything is handled inside the `<img>`. That I guess is the standard HTML output too for an image tag with an `alt` , `src` and `caption`. 

@shon any thoughts?

Reply:
@Pradhvan  If we are not going to add more `special_attrs`, I think we don't  need to iterate on `special_attrs_map` as we have a single item.

Reply:
Agreed 😄